pkg/services/publicdashboards/database/database.go

```diff
@@ -149,118 +152,127 @@ func (d *PublicDashboardStoreImpl) FindByDashboardUid(ctx context.Context, orgId
+func (d *PublicDashboardStoreImpl) ExistsEnabledByDashboardUid(ctx context.Context, dashboardUid string) (bool, error) {
+	hasPublicDashboard := false
+	err := d.sqlStore.WithDbSession(ctx, func(dbSession *db.Session) error {
+		sql := "SELECT COUNT(*) FROM dashboard_public WHERE dashboard_uid=? AND is_enabled=true"
 
-	err := d.sqlStore.WithDbSession(ctx, func(sess *db.Session) error {
-		_, err := sess.UseBool("is_enabled").Insert(&cmd.PublicDashboard)
+		result, err := dbSession.SQL(sql, dashboardUid).Count()
 		if err != nil {
 			return err
 		}
 
-		return nil
+		hasPublicDashboard = result > 0
+		return err
 	})
 
-	return err
+	return hasPublicDashboard, err
 }
 
-// Update updates existing public dashboard
-func (d *PublicDashboardStoreImpl) Update(ctx context.Context, cmd SavePublicDashboardCommand) error {
-	err := d.sqlStore.WithDbSession(ctx, func(sess *db.Session) error {
-		timeSettingsJSON, err := json.Marshal(cmd.PublicDashboard.TimeSettings)
+// ExistsEnabledByAccessToken Responds true if the accessToken exists and the public dashboard is enabled
+func (d *PublicDashboardStoreImpl) ExistsEnabledByAccessToken(ctx context.Context, accessToken string) (bool, error) {
+	hasPublicDashboard := false
+	err := d.sqlStore.WithDbSession(ctx, func(dbSession *db.Session) error {
+		sql := "SELECT COUNT(*) FROM dashboard_public WHERE access_token=? AND is_enabled=true"
+
+		result, err := dbSession.SQL(sql, accessToken).Count()
 		if err != nil {
 			return err
 		}
 
-		_, err = sess.Exec("UPDATE dashboard_public SET is_enabled = ?, annotations_enabled = ?, time_settings = ?, updated_by = ?, updated_at = ? WHERE uid = ?",
-			cmd.PublicDashboard.IsEnabled,
-			cmd.PublicDashboard.AnnotationsEnabled,
-			string(timeSettingsJSON),
-			cmd.PublicDashboard.UpdatedBy,
-			cmd.PublicDashboard.UpdatedAt.UTC().Format("2006-01-02 15:04:05"),
-			cmd.PublicDashboard.Uid)
+		hasPublicDashboard = result > 0
+		return err
+	})
+
+	return hasPublicDashboard, err
+}
+
+// GetOrgIdByAccessToken Returns the public dashboard OrgId if exists and is enabled.
+func (d *PublicDashboardStoreImpl) GetOrgIdByAccessToken(ctx context.Context, accessToken string) (int64, error) {
+	var orgId int64
+	err := d.sqlStore.WithDbSession(ctx, func(dbSession *db.Session) error {
+		sql := "SELECT org_id FROM dashboard_public WHERE access_token=? AND is_enabled=true"
 
+		_, err := dbSession.SQL(sql, accessToken).Get(&orgId)
 		if err != nil {
 			return err
 		}
 
-		return nil
+		return err
 	})
 
-	return err
+	return orgId, err
 }
 
-func (d *PublicDashboardStoreImpl) Delete(ctx context.Context, orgId int64, uid string) (int64, error) {
-	dashboard := &PublicDashboard{OrgId: orgId, Uid: uid}
+// Creates a public dashboard
+func (d *PublicDashboardStoreImpl) Create(ctx context.Context, cmd SavePublicDashboardCommand) (int64, error) {
+	if cmd.PublicDashboard.DashboardUid == "" {
+		return 0, dashboards.ErrDashboardIdentifierNotSet
+	}
```

Comment:
this check should be in the api, as it is a check related to the http request body

Reply:
dashboardUid is non nullable. If someone is searching by an empty uid, we know we can exit early.

check is happening at the api layer as well.

Reply:
so, why are we rechecking here also?